server/index.js

```diff
@@ -57,6 +57,103 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const sql = `
+  select *
+  from "carts"
+  `;
+  db.query(sql)
+    .then(result => {
+      if (!req.session.cartId) {
+        res.json([]);
+      } else {
+        const sql = `
+          select "c"."cartItemId",
+          "c"."price",
+          "p"."productId",
+          "p"."image",
+          "p"."name",
+          "p"."shortDescription"
+          from "cartItems" as "c"
+          join "products" as "p" using ("productId")
+          where "c"."cartId" = $1
+        `;
+        const params = [req.session.cartId];
+        return db.query(sql, params).then(result => res.json(result.rows));
+      }
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId, 10);
+  if (!productId) {
+    next(new ClientError(`productId ${req.body.productId} is not a valid input`, 400));
+    return;
+  }
+  const sql = `
+  select "price"
+  from "products"
+  where "productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        next(new ClientError(`productId ${productId} doesn't exist`), 400);
+        return;
```

Comment:
Ok, here, we have to legit `throw` instead of call `next()` because `next()` and a `return` won't prevent the promise chain from continuing.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw